sdk/datafactory/Azure.ResourceManager.DataFactory/src/Customized/AvroDataset.cs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text.Json;
+using Azure.Core;
+using Azure.Core.Expressions.DataFactory;
+
+namespace Azure.ResourceManager.DataFactory.Models
+{
+    /// <summary> Avro dataset. </summary>
+    [CodeGenSuppress(nameof(AvroDataset), typeof(string), typeof(string), typeof(DataFactoryElement<IList<DatasetDataElement>>),
+        typeof(DataFactoryElement<IDictionary<string, BinaryData>>), typeof(DataFactoryLinkedServiceReference), typeof(IDictionary<string, EntityParameterSpecification>),
```

Comment:
let us use the fully qualified name here for `DataFactoryLinkedServiceReference`.
This should solve the issue.
```suggestion
        typeof(DataFactoryElement<IDictionary<string, BinaryData>>), typeof(Azure.XXX.YYY.ZZZ.DataFactoryLinkedServiceReference), typeof(IDictionary<string, EntityParameterSpecification>),
```